python/adios2/engine.py

```diff
@@ -102,8 +102,16 @@ def put(self, variable, content, mode=bindings.Mode.Deferred):
         """
         if isinstance(content, np.ndarray):
             self.impl.Put(variable.impl, content, mode)
-        else:
+        elif isinstance(content, str):
+            self.impl.Put(variable.impl, content)
+        elif isinstance(content, list):
+            content = np.array(content)
```

Comment:
Why converting any non string list to numpy? It seems like a further copy to another data type.

Reply:
no conversion now